modules/claims_api/app/controllers/claims_api/v1/forms/disability_compensation_controller.rb

```diff
@@ -198,6 +199,21 @@ def validate_form_526_veteran_homelessness!
           end
         end
 
+        def validate_form_526_service_pay!
+          receiving_attr    = form_attributes.dig('servicePay', 'militaryRetiredPay', 'receiving')
+          will_receive_attr = form_attributes.dig('servicePay', 'militaryRetiredPay', 'willReceiveInFuture')
+
+          return if receiving_attr.nil? || will_receive_attr.nil?
+
+          # EVSS does not allow both attributes to be the same value (unless that value is nil)
+          if receiving_attr == will_receive_attr
```

Comment:
nit: i would go with a guard clause here personally. less nesting == easier to read imo

Reply:
I would too, but it makes the line too long. 😞

Reply:
hmmm, it shouldnt...
```
def validate_form_526_service_pay!
  receiving_attr    = form_attributes.dig('servicePay', 'militaryRetiredPay', 'receiving')
  will_receive_attr = form_attributes.dig('servicePay', 'militaryRetiredPay', 'willReceiveInFuture')

  return if receiving_attr.nil? || will_receive_attr.nil?
  return unless receiving_attr == will_receive_attr

  raise ::Common::Exceptions::InvalidFieldValue.new(
    'servicePay.militaryRetiredPay',
    form_attributes['servicePay']['militaryRetiredPay']
  )
end
```
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/GuardClause

Reply:
ahhhh, I was thinking of it in the `raise SomeException if some_condition_true` structure.

Reply:
That is more readable.
Updated [here](https://github.com/department-of-veterans-affairs/vets-api/pull/7647/commits/af9f27807f1f841cc0a9fe626141ac12320299e9).